unison-runtime/src/Unison/Runtime/Machine.hs

```diff
@@ -156,26 +158,31 @@ stk'info s@(BS _ _ sp _) = do
   prn sp
 
 -- Entry point for evaluating a section
-eval0 :: CCache -> ActiveThreads -> Section -> IO ()
+eval0 :: CCache -> ActiveThreads -> RSection -> IO ()
 eval0 !env !activeThreads !co = do
   ustk <- alloc
   bstk <- alloc
+  cmbs <- readTVarIO $ combs env
   (denv, k) <-
-    topDEnv <$> readTVarIO (refTy env) <*> readTVarIO (refTm env)
+    topDEnv cmbs <$> readTVarIO (refTy env) <*> readTVarIO (refTm env)
   eval env denv activeThreads ustk bstk (k KE) dummyRef co
 
 topDEnv ::
+  EnumMap Word64 RCombs ->
   M.Map Reference Word64 ->
   M.Map Reference Word64 ->
   (DEnv, K -> K)
-topDEnv rfTy rfTm
+topDEnv combs rfTy rfTm
   | Just n <- M.lookup exceptionRef rfTy,
+    -- TODO: Should I special-case this raise ref and pass it down from the top rather than always looking it up?
```

Comment:
cc @dolio  What do you think about this particular case?

Reply:
I doubt special casing this will make much difference anywhere. This is essentially just the top level handler for exceptions, which will kill the interpreter with an error message when it's invoked.